.gitmodules

```diff
@@ -6,7 +6,7 @@
 	url = https://github.com/AFLplusplus/Grammar-Mutator
 [submodule "qemu_mode/qemuafl"]
 	path = qemu_mode/qemuafl
-	url = https://github.com/AFLplusplus/qemuafl
+	url = https://github.com/CowBoy4mH3LL/qemuafl
```

Comment:
No :-) if this needs qemuafl changes then put them as a PR to our qemuafl repo

Reply:
Go it -- so I'll put up another PR qemuafl

Reply:
Yes imho that is all that is needed.

Reply:
I understand Dominik’s opinion on this but the user base for afl++ is currently and will be still for at least a year or longer much much larger than that of libafl

Reply:
I'm not talking about AFL++ vs LibAFL, I'm talking about qemuafl vs libafl_qemu, the former isn't really actively maintained, adding an afl++ forkserver to the latter would be the better time invest

Reply:
https://github.com/AFLplusplus/LibAFL/pull/1983

Reply:
> I'm not talking about AFL++ vs LibAFL, I'm talking about qemuafl vs libafl_qemu, the former isn't really actively maintained, adding an afl++ forkserver to the latter would be the better time invest

Ah sorry i misunderstood! I totally agree. Is someone on it? Adding a forkserver is easy but supporting the afl qemu envs are likely quite some work

Reply:
I suspect even if we move away from qemuafl, since libafl_qemu builds on QEMU itself, we can integrate this hooking system into libafl_qemu as well -- infact, going forward I am thinking of doing a quick performance comparison of the two hooking methods -- if the hooking system of this PR comes out to be faster (due to native integration into QEMU) we can keep both

Reply:
@domenukk -- absolutely, I'll be on it sooner

@vanhauser-thc -- is it OK to create qemuafl PR on master directly? -- I don't see a dev branch

Reply:
Yes make a pr to master 👍🏻

Reply:
Deal!!

Reply:
> I would assume libafl_qemu native hooks will be _a lot_ faster since they are optimized for performance, but I would be very keen to see the benchmark! :)

Saw that in the paper -- but how are hooks called? Do you follow instructions or TCGs, or leverage the plugin ecosystem or is it event based like trap in our case -- Just out of curiosity

Reply:
@CowBoy4mH3LL you still need to reverse this

Reply:
Sorry about that -- reset it now :)